File: Data Structures & Algorithms/longest-increasing-subsequence/submission-41.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        

        def dfs(last, i): 
            if i == len(nums): 
                return 0 
            
            res = dfs(last, i + 1) 

            if last == None or nums[i] > last: 
                res = max(res, 1 + dfs(nums[i], i + 1)) 

            return res 

        return dfs(None, 0)    

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-41.js
class Solution {
    /**
     * @param {string} s
     * @param {number} k
     * @return {number}
     */
    characterReplacement(s, k) {  
        let count = {}
        let res = 0

        let l = 0 
        for (let r = 0; r < s.length; r ++) {
            let cur = s[r] 

            count[cur] = ( count[cur] || 0 ) + 1 

            while(((r - l + 1) - Math.max(...Object.values(count))) > k) {
                let deleted = s[l] 
                count[deleted] -- 
                l ++
            } 

            
            res = Math.max(res, (r - l + 1))
        } 

        return res
    }
}
